.travis.yml

```diff
@@ -1,61 +1,137 @@
   fast_finish: true
 
+# Use this to prepare the system to install prerequisites or dependencies.
+# e.g. sudo apt-get update.
+# Failures in this section will result in build status 'errored'.
+before_install:
+
+# Use this to prepare your build for testing.
+# e.g. copy database configurations, environment variables, etc.
+# Failures in this section will result in build status 'errored'.
 before_script:
+  # Turn off Xdebug. See https://core.trac.wordpress.org/changeset/40138.
+  - phpenv config-rm xdebug.ini || echo "Xdebug not available"
+
   - export PATH="$HOME/.composer/vendor/bin:$PATH"
+
+  # Couple the PHPUnit version to the PHP version.
   - |
-      if [[ ${TRAVIS_PHP_VERSION:0:2} == "7." ]]; then
-        composer global require "phpunit/phpunit=5.7.*"
-      else
+    case "$TRAVIS_PHP_VERSION" in
+      7.*)
+        echo "Using PHPUnit 6.1"
+        composer global require "phpunit/phpunit=6.1.*"
+        ;;
+      *)
+        echo "Using PHPUnit 4.8"
         composer global require "phpunit/phpunit=4.8.*"
-      fi
-  - composer install
-  - |
-    if [[ ! -z "$WP_VERSION" ]] ; then
-      bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION
-    fi
+        ;;
+    esac
+
+  - og_dir="$(pwd)"
+
+  # Set up WordPress installation.
+  - export WP_DEVELOP_DIR=/tmp/wordpress
+  - export WP_TESTS_DIR=${WP_DEVELOP_DIR}/tests/phpunit
+  - export WP_CORE_DIR=${WP_DEVELOP_DIR}/src/
+  - mkdir -p $WP_DEVELOP_DIR
+
+  # Use the Git mirror of WordPress.
   - |
-    if [[ "$WP_TRAVISCI" == "phpcs" ]] ; then
-      npm install
-    fi
+    case "$WP_VERSION" in
+      latest)
+        curl -s http://api.wordpress.org/core/version-check/1.7/ > /tmp/wp-latest.json
+        grep '[0-9]+\.[0-9]+(\.[0-9]+)?' /tmp/wp-latest.json
+        LATEST_VERSION=$(grep -o '"version":"[^"]*' /tmp/wp-latest.json | sed 's/"version":"//')
+        if [[ -z "$LATEST_VERSION" ]]; then
+          echo "Latest WordPress version could not be found."
+          exit 1
+        fi
+        git clone --depth=1 --branch="$LATEST_VERSION" git://develop.git.wordpress.org/ $WP_DEVELOP_DIR
+        ;;
+      *)
+        git clone --depth=1 --branch="$WP_VERSION" git://develop.git.wordpress.org/ $WP_DEVELOP_DIR
+        ;;
+    esac
+
+  # Set up WordPress configuration.
+  - cd $WP_DEVELOP_DIR
+  - echo $WP_DEVELOP_DIR
+  - cp wp-tests-config-sample.php wp-tests-config.php
+  - sed -i "s/youremptytestdbnamehere/wordpress_test/" wp-tests-config.php
+  - sed -i "s/yourusernamehere/root/" wp-tests-config.php
+  - sed -i "s/yourpasswordhere//" wp-tests-config.php
+  - echo "define( 'JETPACK_DEV_DEBUG', true );" >> wp-tests-config.php
+  - cat wp-tests-config.php
 
+  # Create WordPress database.
+  - mysql -e 'CREATE DATABASE wordpress_test;' -uroot
+
+  # Set up the plugin.
+  - cd $og_dir
+  - mkdir -p "${WP_CORE_DIR}wp-content/plugins/apple-news"
+  - cp -R . "${WP_CORE_DIR}wp-content/plugins/apple-news/"
```

Comment:
why is this block necessary? that is, why is it necessary that the plugin live in that directory?

Reply:
It's possible that it doesn't need to. I was having some difficulty getting the tests to run properly, and this is one of the steps I took to get everything working. I'll experiment with pulling it out.

Reply:
As it turns out, I didn't need this after all. Thanks!